src/backend/dx11/src/lib.rs

```diff
@@ -3869,25 +3916,14 @@ impl From<pso::DescriptorType> for DescriptorContent {
                 ty: Bdt::Uniform, ..
             } => DescriptorContent::CBV,
             Dt::Buffer {
-                ty: Bdt::Storage { read_only: true },
```

Comment:
can you remind me, what's the problem with binding read-only storage as SRV?
I thought your issue was only the case where we wanted to bind the same resource as both UAV and SRV. But in this case, it would be either one but not both.

Reply:
I think there is a better solution hiding here. We have a place where we connect the shader AST with the binding layout, it's called `patch_spirv_resources` in DX11 backend. There, if we see that the shader has `NonWritable` but the layout isn't read-only, we could just remove the `NonWritable` declaration. The benefit of doing so is that we'd still go a more efficient SRV path for truly read-only bindings, which WebGPU would use.